core/src/main/java/io/grpc/ExponentialBackoffPolicy.java

```diff
@@ -0,0 +1,150 @@
+    @Override
+    public BackoffPolicy get() {
+      return ExponentialBackoffPolicy.builder()
+          .initialBackoffMilis(1, TimeUnit.SECONDS)
+          .maxBackoffMillis(2, TimeUnit.MINUTES)
+          .multiplier(1.6)
+          .jitter(.2)
+          .build();
+    }
+  }
+
+  private final Random random;
+  private final long initialBackoffMillis;
+  private final long maxBackoffMillis;
+  private final double multiplier;
+  private final double jitter;
+
+  private long nextBackoffMillis;
+
+  private ExponentialBackoffPolicy(Random random, long initialBackoffMilis, long maxBackoffMillis,
+                                   double multiplier, double jitter) {
+    this.random = random;
+    this.initialBackoffMillis = initialBackoffMilis;
+    this.maxBackoffMillis = maxBackoffMillis;
+    this.multiplier = multiplier;
+    this.jitter = jitter;
+
+    this.nextBackoffMillis = initialBackoffMillis;
+  }
+
+  @Override
+  public long nextBackoffMillis() {
+    long currentBackoffMillis = nextBackoffMillis;
+    nextBackoffMillis = Math.min((long) (currentBackoffMillis * multiplier), maxBackoffMillis);
+    return currentBackoffMillis
+            + uniformRandom(-jitter * currentBackoffMillis, jitter * currentBackoffMillis);
+  }
+
+  private long uniformRandom(double low, double high) {
+    checkArgument(high >= low);
+    double mag = high - low;
+    return (long) (random.nextDouble() * mag + low);
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static final class Builder {
+    private Random random;
+    private long initialBackoffMilis;
+    private long maxBackoffMillis;
+    private double multiplier;
+    private double jitter;
+
+    private Builder() {
+    }
+
+    public Builder random(Random random) {
+      this.random = random;
+      return this;
+    }
+
+    public Builder initialBackoffMilis(long initialBackoff, TimeUnit timeUnit) {
+      this.initialBackoffMilis = timeUnit.toMillis(initialBackoff);
+      return this;
+    }
+
+    public Builder maxBackoffMillis(long maxBackoff, TimeUnit timeUnit) {
+      this.maxBackoffMillis = timeUnit.toMillis(maxBackoff);
+      return this;
+    }
+
+    public Builder multiplier(double multiplier) {
+      this.multiplier = multiplier;
+      return this;
+    }
+
+    public Builder jitter(double jitter) {
+      this.jitter = jitter;
+      return this;
+    }
+
+    /**
+     * Builds {@link BackoffPolicy} instance.
+     *
+     * @return BackoffPolicy object.
+     */
+    public ExponentialBackoffPolicy build() {
+      checkArgument(initialBackoffMilis > 0, "initialBackoffMilis must be greater than zero");
+      checkArgument(maxBackoffMillis > 0, "maxBackoffMillis must be greater than zero");
+      checkArgument(initialBackoffMilis < maxBackoffMillis, "initialBackoffMilis can't be greater"
+              + "than maxBackoffMillis");
+      checkArgument(multiplier > 0, "multiplier must be greater than zero");
```

Comment:
I would personally find it very confusing to find out that `ExponentialBackoffPolicy` is linear backoff so my preference is to not allow it. We can always provide `LinearBackoffPolicy` but I wouldn't be mixing them together (cause allowing 0 here would effectively mean that).